_gsocproposals/2020/proposal_RucioSupportBot.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Support for Rucio Users with Native Language Processing
+layout: gsoc_proposal
+project: Rucio
+year: 2020
+organization:
+    - CERN
+---
+
+## Description
+
+Rucio is an open source software framework that provides scientific collaborations with the functionality to organize, manage, and access their volumes of data. Data in Rucio is organized using Data Identifiers (DIDs) which have three levels of granularity viz.: files, datasets and containers, respectively. Datasets are used to group a set of files to facilitate bulk operations on them eg. in transfers, deletions, and organizational purposes. Users are permitted to perform certain actions on the DIDs such as downloads, uploads, transfers, etc. Expert support, through different escalation levels, is offered to the users in case of problems. In case when satisfying answers were not found at previous support levels, a request from a user or a group of users can reach the Rucio support. Due to the vast amount of support requests, we are looking into methods to assist the support team in answering these requests. Ideally, the support would be provided by an intelligent bot able to process and understand the user’s requests and finally trigger appropriate action. 
+The Native Language Processing (NLP) is fairly developed technique used nowadays in many fields with need of analyzing large chunks of text and automate actions on them. Next to a few other use cases in Rucio project, an important one is to process responses and questions from users and to provide satisfying answers. That could be achieved by development of a prototype of a bot that is capable to handle the user's requests which don't exceed certain complexity. The request would be handled by experts only in case when the bot can not provide a reliable answer. Such a complex task consists of the following milestones.
+
+## Milestones for GSoC Student
+
+ * Input Data: 
+   * Creation of corpora or use of existing one.
```

Comment:
Yes. There are several available corporas of texts that are freely available for NLP, e.g. few pages from almost every book in classical literature. We would need to create one dedicated to data management because there's not any and we can not allow the student to access the emails directly (permission policy).